### PR TITLE
upgrade ansible sample to use sdk 0.16

### PR DESCRIPTION
**Pull request by camilamacedo86**
**Description**
Full scaffold upgrade with all 0.16 changes

**Motivativation**
SDK release 

**Local Test**
Screenshot 2020-03-19 at 17 50 05

### CODE REVIEW

File: ansible/memcached-operator/molecule/test-local/molecule.yml [hunk 1]
```diff
@@ -6,21 +6,26 @@ driver:
...
-  override_command: no
-  exposed_ports:
-    - 8443/tcp
-    - 10080/tcp
-  published_ports:
-    - 0.0.0.0:${TEST_CLUSTER_PORT:-10443}:8443/tcp
-  pre_build_image: yes
-  volumes:
-    - ${MOLECULE_PROJECT_DIRECTORY}:/build:Z
+  - name: kind-test-local
```
Comment (camilamacedo86) [inline ansible/memcached-operator/molecule/test-local/molecule.yml]: @fabianvf all these changes scaffold X samples in the molecule are not in the migration guide. 
Did you make them in this release or it was outdated? Do you know it?

**Comment by bharathi-tenneti** [comment]
@camilamacedo86 From 0.16.0 release, I see that Ansible logs will be displayed in one log. However, see below:
```
btenneti@Bharathis-MacBook-Pro ~/go-workspace/src/github.com/bharathi-tenneti/PR-reviews/operator-sdk-samples/ansible/memcached-operator (up-go-0.16) $ kubectl logs deployment.apps/memcached-operator -n=memcached
error: a container name must be specified for pod memcached-operator-78dcccc869-4xw45, choose one of: [ansible operator]
btenneti@Bharathis-MacBook-Pro ~/go-workspace/src/github.com/bharathi-tenneti/PR-reviews/operator-sdk-samples/ansible/me
```

**Comment by bharathi-tenneti** [comment]
Also, below error for logs.
```
btenneti@Bharathis-MacBook-Pro ~/go-workspace/src/github.com/bharathi-tenneti/PR-reviews/operator-sdk-samples/ansible/memcached-operator (up-go-0.16) $ kubectl logs deployment.apps/memcached-operator -n=memcached -c operator
{"level":"info","ts":1584731266.5927854,"logger":"cmd","msg":"Go Version: go1.13.8"}
{"level":"info","ts":1584731266.5931005,"logger":"cmd","msg":"Go OS/Arch: linux/amd64"}
{"level":"info","ts":1584731266.5935078,"logger":"cmd","msg":"Version of operator-sdk: v0.15.2"}
{"level":"info","ts":1584731266.5937603,"logger":"cmd","msg":"Watching namespace.","Namespace":"memcached"}
{"level":"info","ts":1584731266.9533858,"logger":"controller-runtime.metrics","msg":"metrics server is starting to listen","addr":"0.0.0.0:8383"}
{"level":"info","ts":1584731266.9539287,"logger":"watches","msg":"Environment variable not set; using default value","envVar":"WORKER_MEMCACHED_CACHE_EXAMPLE_COM","default":1}
{"level":"info","ts":1584731266.954035,"logger":"watches","msg":"Environment variable not set; using default value","envVar":"ANSIBLE_VERBOSITY_MEMCACHED_CACHE_EXAMPLE_COM","default":2}
{"level":"error","ts":1584731266.95407,"logger":"watches","msg":"Invalid ansible path for GVK: cache.example.com/v1alpha1, Kind=Memcached","error":"role path must be absolute","stacktrace":"github.com/go-logr/zapr.(*zapLogger).Error\n\tpkg/mod/github.com/go-logr/zapr@v0.1.1/zapr.go:128\ngithub.com/operator-framework/operator-sdk/pkg/ansible/watches.(*Watch).Validate\n\tsrc/github.com/operator-framework/operator-sdk/pkg/ansible/watches/watches.go:150\ngithub.com/operator-framework/operator-sdk/pkg/ansible/watches.Load\n\tsrc/github.com/operator-framework/operator-sdk/pkg/ansible/watches/watches.go:215\ngithub.com/operator-framework/operator-sdk/pkg/ansible.Run\n\tsrc/github.com/operator-framework/operator-sdk/pkg/ansible/run.go:104\ngithub.com/operator-framework/operator-sdk/cmd/operator-sdk/execentrypoint.newRunAnsibleCmd.func1\n\tsrc/github.com/operator-framework/operator-sdk/cmd/operator-sdk/execentrypoint/ansible.go:38\ngithub.com/spf13/cobra.(*Command).execute\n\tpkg/mod/github.com/spf13/cobra@v0.0.5/command.go:826\ngithub.com/spf13/cobra.(*Command).ExecuteC\n\tpkg/mod/github.com/spf13/cobra@v0.0.5/command.go:914\ngithub.com/spf13/cobra.(*Command).Execute\n\tpkg/mod/github.com/spf13/cobra@v0.0.5/command.go:864\nmain.main\n\tsrc/github.com/operator-framework/operator-sdk/cmd/operator-sdk/main.go:39\nruntime.main\n\t/home/travis/.gimme/versions/go1.13.8.linux.amd64/src/runtime/proc.go:203"}
```

**Comment by camilamacedo86** [comment]
@bharathi-tenneti, 

I am quite sure that you are not testing the PR. Note that it has no longer the -c operator: https://github.com/operator-framework/operator-sdk-samples/pull/118/files#diff-c30191b6578e03cdc7b0f9d1788ca91fR18.

To do the test you need pull the PR and run`make install`.

File: ansible/memcached-operator/deploy/operator.yaml [hunk 2]
```diff
@@ -14,21 +15,9 @@ spec:
...
-          image: "REPLACE_IMAGE"
-          imagePullPolicy: "Always"
-          volumeMounts:
-          - mountPath: /tmp/ansible-operator/runner
-            name: runner
-            readOnly: true
-        - name: operator
-          # Replace this with the built image name
-          image: "REPLACE_IMAGE"
+          image: "IMAGE"
```
Comment (camilamacedo86) [inline ansible/memcached-operator/deploy/operator.yaml]:
```suggestion
          image: "REPLACE_IMAGE"
```

File: ansible/memcached-operator/molecule/test-local/molecule.yml (hunk 1 above)
Comment (fabianvf) [reply ansible/memcached-operator/molecule/test-local/molecule.yml]: None of the changes to the molecule scenarios are strictly necessary, the new stuff is just cleaner, though I should probably add a section touching on the bigger changes

Comment (camilamacedo86) [reply ansible/memcached-operator/molecule/test-local/molecule.yml]: I think we can add something in the migration doc to share the changes anyway. We usually put (optional) and describe how to get that. However, really tks for letting me know and your review.

**Pull request merged**